CNV#41400 : Storage checkup is missing UI documentation

aspauldi opened the pull request:
Version(s): 4.15+

Issue: [CNV-41400](https://issues.redhat.com/browse/CNV-41400)

Link to docs preview: https://79006--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/monitoring/virt-running-cluster-checkups.html

QE review:
- [x] QE has approved this change.

aspauldi commented:
/label cnv

aspauldi commented:
Hi @arnongilboa, As I mentioned I wanted to do in the Jira, I moved the procedure for running a storage checkup on the CLI up in the TOC, just after the same procedure for running a latency checkup. Can you do another look at the TOC and verify that it's okay? Thanks!

arnongilboa commented:
> Hi @arnongilboa, As I mentioned doing in the [Jira](https://issues.redhat.com/browse/CNV-41400), I moved the procedure for running a storage checkup on the CLI up in the TOC, just after the same procedure for running a latency checkup. Can you take another look at the TOC and verify that it's okay? Thanks!

The TOC looks ok. Few minor comments:
1. Shorten the title "Running latency and storage cluster checkups in the web console" to "Running checkups in the web console"
1. Move the (!) section to the latency checkup
1. In the storage checkup section maybe just mention `for additional information on the checkup configuration and results, refer the CLI section`
1. Shorten the title "Running latency and storage cluster checkups on the CLI" to "Running checkups on the CLI"
1. Consider moving the DPDK to the CLI section, as that's what it is

aspauldi commented:
Thanks, @arnongilboa. What do you mean by "Move the (!) section to the latency checkup"?

arnongilboa commented:
> Thanks, @arnongilboa. Regarding your comments:
> 
> Comment 2: What do you mean by "Move the (!) section to the latency checkup"?
Simply meant moving the warning/comment box down down to its relevant section

> Comment 3: Our documentation style/guidelines do not permit adding a cross reference from a module (for ex., a procedure, concept topic, reference topic) so I can't add the cross reference you mentioned.

Just meant to mention it without a link, but it's not a must.

> 
> Your remaining comments have all been incorporated. Can you take another look? Thanks.

aspauldi commented:
Does the warning that you suggest moving down apply specifically to running the latency checkup in the web console, from the CLI, or both? I'm thinking that it's both, and I might move the info out of the warning and into the prerequisites for each method if so.

arnongilboa commented:
> @arnongilboa, Does the warning that you suggest moving down apply specifically to running the latency checkup in the web console, from the CLI, or both? I'm thinking that it's both, and I might move the info out of the warning and into the prerequisites for each method if so.

I guess it's relevant for both web console and UI, but I'm from the CNV storage.
@ormergi from CNV network may better help here.

ormergi commented:
> @arnongilboa, Does the warning that you suggest moving down apply specifically to running the latency checkup in the web console, from the CLI, or both? I'm thinking that it's both, and I might move the info out of the warning and into the prerequisites for each method if so.

Hi, the requirement for the latency checkup (having a bridge on the nodes) is required whether you run the checkup through CLI or web console.

aspauldi commented:
Thanks, @ormergi and @arnongilboa. I believe I've addressed all your comments. If so, can you indicate your approval of the PR so that I can send it on to QE? 

(Note that our doc guidelines permit the inclusion of crossreferences only in assemblies and not in modules, such as procedures. For this reason, I moved the warning about creating a bridge interface before doing a latency checkup to the beginning of the topic.)

arnongilboa commented:
@aspauldi not the `Example input config map` data and `Example output config map` status fields are a bit outdated. There were some renames/additions, so doc needs minor fixes, see:
https://github.com/kiagnose/kubevirt-storage-checkup?tab=readme-ov-file#checkup-results-retrieval

aspauldi commented:
@arnongilboa , Just checking before I make the updates I think that you want.

In **Running a storage checkout on the CLI**>**Example input config map and job manifest**, I see that I can add **spec.param.storageClass** and **spec.param.vmiTimeout** after **spec.timeout**. Is that correct? What would be a suitable value for **spec.param.storageClass**? 

In **Running a storage checkout on the CLI**>**Example output config map (success)**, I see a list of statuses. Some of them do not appear in the link you provided (https://github.com/kiagnose/kubevirt-storage-checkup?tab=readme-ov-file#checkup-results-retrieval). I can add/remove statuses to match the info in the link, but that would mean removing, for example, **status.result.cnvVersion: 4.16.2**, which appears to have been added to the list just about a month ago by another writer. Can you clarify for me? 

Thanks!

arnongilboa commented:
> In **[Running a storage checkout on the CLI](https://79006--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/monitoring/virt-running-cluster-checkups.html#virt-checking-storage-configuration_virt-running-cluster-checkups)**>**Step 3: Example input config map and job manifest**, I see that I can add **spec.param.storageClass** and **spec.param.vmiTimeout** after **spec.timeout**. Is that correct? What would be a suitable value for **spec.param.storageClass**?

any available storage class, e.g. `ocs-storagecluster-ceph-rbd-virtualization`

> 
> In **[Running a storage checkout on the CLI](https://79006--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/monitoring/virt-running-cluster-checkups.html#virt-checking-storage-configuration_virt-running-cluster-checkups)**>**Step 6: Example output config map (success)**, I see a list of statuses. Some of them do not appear in the link you provided (https://github.com/kiagnose/kubevirt-storage-checkup?tab=readme-ov-file#checkup-results-retrieval). I can add/remove statuses to match the info in the link, but that would mean removing, for example, **status.result.cnvVersion: 4.16.2**, which appears to have been added to the list just about a month ago by another writer. Can you clarify for me?

I somehow missed documenting the cnvVersion & ocpVersion upstream. Keep them. The rest are additions and renames  (storageWith... -> storageProfilesWith...).

Thanks!

aspauldi commented:
Hi @arnongilboa , I think I've got all the renames in **Example output config map (success)**. I also tried adding callouts if they were missing, but I'm not sure they are all correct. Can you review them? Thanks!

aspauldi commented:
Hi @arnongilboa, Can you take a look at the recent updates to this doc PR? Thanks

arnongilboa commented:
> Hi @arnongilboa, Can you take a look at the recent updates to this doc PR? Thanks

/lgtm

aspauldi commented:
Awaiting QE review

hmeir commented on modules/virt-checking-storage-configuration.adoc:
```diff
@@ -166,15 +167,17 @@ data:
+  status.result.goldenImagesNotUpToDate: <data_import_cron_list> # <8>
+  status.result.ocpVersion: 4.16.0 # <9>
+  status.result.pvcBound: "true" # <10>
+  status.result.storageProfileMissingVolumeSnapshotClass: <storage_class_list> # <11>
```
<storage_profile_list>

arnongilboa replied:
Right, but storage classes and their profiles has the same name so not sure how important is it.

aspauldi replied:
I'll change this to <storage_profile_list>

hmeir commented on modules/virt-checking-storage-configuration.adoc:
```diff
@@ -188,19 +191,25 @@ data:
-<9> The list of virtual machines that use the Ceph RBD storage class when the virtualization storage class exists.
-<10> The list of virtual machines that use an Elastic File Store (EFS) storage class where the GID and UID are not set in the storage class.
+<5> The {VirtProductName} version.
+<6> Specifies if there is a default storage class.
```
Its actually printing the default storage class, and it can be more then one. Its not a boolean but a list of default storageclasses, if exists.
@arnongilboa correct me If i'm wrong.

arnongilboa replied:
We currently look for only a single default storage class and error if there is none or there are more than one.
We don't support yet virt default storage class in the checks, and should add it soon.

hmeir commented on modules/virt-checking-storage-configuration.adoc:
```diff
@@ -188,19 +191,25 @@ data:
+<5> The {VirtProductName} version.
+<6> Specifies if there is a default storage class.
+<7> The list of golden images whose data source is not ready.
+<8> The list of golden images whose data import cron is not up-to-date.
```
another condition is when its datasource is not ready

arnongilboa replied:
We already have it in <7> so there is a coverage.

hmeir commented on modules/virt-checking-storage-configuration.adoc:
```diff
@@ -188,19 +191,25 @@ data:
-<10> The list of virtual machines that use an Elastic File Store (EFS) storage class where the GID and UID are not set in the storage class.
+<5> The {VirtProductName} version.
+<6> Specifies if there is a default storage class.
+<7> The list of golden images whose data source is not ready.
```
Its also for when the datasource has no boot source(pvc/snapshot)

arnongilboa replied:
Right, but not sure it's needed to be added as it's not the common case.

aspauldi commented:
Hi, I have kept the content as agreed upon with @arnongilboa , based on his responses to your concerns, @hmeir. If that's acceptable to you, can you please give your approval for this PR, @hmeir? Thank you both so much.

hmeir commented:
> Hi, I have kept the content as agreed upon with @arnongilboa , based on his responses to your concerns, @hmeir. If that's acceptable to you, can you please give your approval for this PR, @hmeir? Thank you both so much.

/lgtm

aspauldi commented:
/label peer-review-needed

skopacz1 commented on virt/monitoring/virt-running-cluster-checkups.adoc:
```diff
@@ -22,28 +25,28 @@ include::snippets/technology-preview.adoc[]
 
 include::modules/virt-about-cluster-checkup-framework.adoc[leveloffset=+1]
 
```
There needs to be some sort of ID above this heading, here's one option:
```suggestion
[id="virt-running-cluster-checkups-console_{context}"]
```

aspauldi replied:
Thanks!

skopacz1 commented on virt/monitoring/virt-running-cluster-checkups.adoc:
```diff
@@ -22,28 +25,28 @@ include::snippets/technology-preview.adoc[]
-include::modules/virt-measuring-latency-vm-secondary-network.adoc[leveloffset=+1]
 
-include::modules/virt-checking-cluster-dpdk-readiness.adoc[leveloffset=+1]
+== Running checkups on the CLI
```
There also needs to be an ID above this heading:
```suggestion
[id="virt-running-cluster-checkups-cli_{context}"]
== Running checkups on the CLI
```

aspauldi replied:
Thanks here, too!

skopacz1 commented on virt/monitoring/virt-running-cluster-checkups.adoc:
```diff
@@ -22,28 +25,28 @@ include::snippets/technology-preview.adoc[]
+== Running checkups on the CLI
 
-include::modules/virt-dpdk-config-map-parameters.adoc[leveloffset=+2]
+Use the following procedures the first time you run checkups on the CLI.
```
This may be out of scope for this PR, but is there any documentation for how to run additional checkups on the CLI, similar to the section for the web console?

aspauldi replied:
Not that I am aware of, no.

skopacz1 commented on virt/monitoring/virt-running-cluster-checkups.adoc:
```diff
@@ -22,28 +25,28 @@ include::snippets/technology-preview.adoc[]
+
+include::modules/virt-dpdk-config-map-parameters.adoc[leveloffset=+3]
+
+include::modules/virt-building-vm-containerdisk-image.adoc[leveloffset=+3]
```
Is this specific section meant to have a +3 level offset? It was previously a +2. Right now, it seems like it's a subsection of "Running a DPDK checkup on the CLI".

If that's not intentional, I'd change the level offset to what it should be, whether that's a +2 or a +1. Otherwise this comment can be ignored

aspauldi replied:
If I understand correctly what you are asking, I believe the offsets are correct. The TOC looks like this:

▪ About the OpenShift Virtualization cluster checkup framework
▪ Running checkups in the web console
     ▪ Running a latency checkup in the web console
     ▪ Running a storage checkup in the web console
▪ Running checkups on the CLI
     ▪ Running a latency checkup on the CLI
     ▪ Running a storage checkup on the CLI
     ▪ Running a DPDK checkup on the CLI
▪ Additional resources

aspauldi commented on modules/virt-checking-cluster-dpdk-readiness.adoc:
```diff
@@ -4,7 +4,7 @@
 :_mod-docs-content-type: PROCEDURE
 [id="virt-checking-cluster-dpdk-readiness_{context}"]
-= DPDK checkup
+= Running a DPDK checkup on the CLI
```
Changed headings to " . . . by using the command line"

aspauldi commented on virt/monitoring/virt-running-cluster-checkups.adoc:
```diff
@@ -22,28 +25,28 @@ include::snippets/technology-preview.adoc[]
-====
-Before you run a latency checkup, you must first xref:../../virt/vm_networking/virt-connecting-vm-to-linux-bridge.adoc#virt-connecting-vm-to-linux-bridge[create a bridge interface] on the cluster nodes to connect the VM's secondary interface to any interface on the node. If you do not create a bridge interface, the VMs will not start and the job will fail.
-====
+Use the following procedures the first time you run checkups in the web console. For additional checkups, click *Run checkup* on either checkup tab, and select the appropriate checkup from the drop down menu.
```
I agree that it could be positioned better than it is. I wish we could create xrefs from modules. I'll leave as is for now.

aspauldi commented:
/label merge-review-needed

bergerhoffer commented:
/label merge-review-in-progress

bergerhoffer reviewed (commented):
LGTM!

bergerhoffer commented:
/cherrypick enterprise-4.17

bergerhoffer commented:
/cherrypick enterprise-4.16

bergerhoffer commented:
/cherrypick enterprise-4.15

bergerhoffer commented:
@aspauldi The CP to 4.15 failed, can you please open a manual CP and send it to merge review squad?
